### DISCUSSION TITLE
consult hangs with find-file at emacs startup

### DISCUSSION THREAD

**Discussion by rickswe** [minad/consult#936] [Q&A] [answered]
Hello,

Was something changed recently that affects this code? It takes a while to bring up the results at
startup. After it hangs for a while then I'm able to search normally. This was not a problem a few weeks ago.

```lisp
(defun my/fd (&optional dir flags initial)
    (interactive "P")
    (setq my/fd-flags (if flags (append (split-string my/fd-default)(split-string flags))(split-string my/fd-default)))
    (let* ((prompt-dir (consult--directory-prompt "Fd" dir))
          (default-directory (car (last prompt-dir))))
      (find-file (consult--find (car prompt-dir) #'consult--fd-builder initial))))
```

I usually call it like this:

```lisp
(defun my/fd-org () "org/"       (interactive)(my/fd (concat org-directory) "--extension=org --extension=md --extension=tex" ))
```

Thanks

**Comment by minad** [comment]
Did you try consult-fd, which has been added a short while ago?

**Reply by rickswe** [reply, marked as answer]
If it was added over 3 months ago I probably considered it at one point.
I figured out what the problem was. My code is working now so if the
above code is working now should I incoporate `consult-fd`? Or leave
it alone cause it's working now?

**Reply by oantolin** [reply]
Out of curiosity, what was the problem?

**Reply by rickswe** [reply]
Hello Omar,

It was a bash script doing a recursive call to fd. I found out about it when
I was getting strange errors when using `fd`. I deleted the script that was
causing the problem.

I still haven't figured out [Embark consult-line issue 637](https://github.com/oantolin/embark/issues/637). :( 
Usually I just use vertico-exit and move on.
